src/bucklescript/graphql_ppx.re

```diff
@@ -96,7 +96,14 @@ let rewrite_query = (~schema=?, ~loc, ~delim, ~query, ()) => {
         /*  the only call site of schema, make it lazy! */
         schema: Lazy.force(Read_schema.get_schema(schema)),
       };
-      switch (Validations.run_validators(config, document)) {
+      let (errors, warnings) = Validations.run_validators(config, document);
+      warnings
+      |> List.iter(((loc, msg)) => {
```

Comment:
This wouldn't work. `List.iter` expexts unit but you're retuning expression here.

Another thing is - to show a warning to a user we have to add it to `Pmod_structure`.
There is a difference between errors and a valid query. I think in warning case you have to construct
`Ast_mapper.attribute_of_warning` (http://caml.inria.fr/pub/docs/manual-ocaml/compilerlibref/Ast_mapper.html) expression and add it to both valid and invalid module to show warnings all the time.